prusti-viper/src/encoder/mir/procedures/encoder/mod.rs

```diff
@@ -829,22 +867,54 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         operand: &mir::Operand<'tcx>,
     ) -> SpannedEncodingResult<()> {
         let span = self.encoder.get_span_of_location(self.mir, location);
+
+        let deref_base: Option<vir_high::Expression> = match &encoded_target {
+            vir_high::Expression::Deref(vir_high::Deref { box base, .. }) => Some(base.clone()),
+            _ => None,
+        };
+        if let Some(base) = &deref_base {
+            self.encode_open_mut_ref(block_builder, location, base, encoded_target.clone())?;
```

Comment:
It seems that in all cases you wrap the calls in `if` statements. I think the code would be more readable if the `if` was inside the function.

Reply:
agree, done